static/js/teams.js

```diff
@@ -41,35 +53,39 @@ function changeTeamScores(movedElement, movedElementParentId, targetElementId){
     if (targetElementId != 'leftover_participants'){
         let teamScoreSpan = document.getElementById(targetElementId +'_score');
         teamScoreSpan.innerText = parseInt(teamScoreSpan.innerText) + movedLevel;
-    }
-
-    if(movedElementParentId.includes('team')) {
         let removeScoreSpan = document.getElementById(movedElementParentId +'_score');
-        removeScoreSpan.innerText = parseInt(removeScoreSpan.innerText) - movedLevel;
+        if(removeScoreSpan != null) {
+            removeScoreSpan.innerText = parseInt(removeScoreSpan.innerText) - movedLevel;
+        }
     }
 }
 
 function changeTeamData(movedElement, movedElementParentId, targetElementId){
     /* Removes the participant from the team the participant was previously 
     part of and adds it to the new team in the team data object which is used
     to create or edit the teams */
-    let team = movedElementParentId.includes('team')
-                ? teams[movedElementParentId]
-                : leftover_participants;
-    let targetTeam = targetElementId.includes('team')
-                    ? teams[targetElementId]
-                    : leftover_participants;
+    if (movedElementParentId == targetElementId) {
+        return
+    }
+    let movedElementParent= document.getElementById(movedElementParentId);
+    console.log(movedElementParent)
+
+    let team = movedElementParentId.includes('leftover_participants')
+                ? leftover_participants
+                : teams[movedElementParentId];
+    let targetTeam = targetElementId.includes('leftover_participants')
+                    ? leftover_participants
+                    : teams[targetElementId];
     let userid = movedElement.dataset.userid;
     let user = team.filter(x => x.userid == userid)[0];
-    targetTeam.push(user)
-
-    if(movedElementParentId.includes('team')){
-        teams[movedElementParentId] = teams[movedElementParentId]
-                                        .filter(x => x.userid != userid);
-    } else {
+    if(movedElementParentId.includes('leftover_participants')){
```

Comment:
be mindful of code formatting -- some spaces missing throughout the file, like a couple on this line for example...

Reply:
Not sure exactly how this should be indented, so I moved it back to a one-liner. Should be easier to read now.